uniswap-v2/src/modules/21_map_events_entity.rs

```diff
@@ -0,0 +1,206 @@
+                &event,
+                &swap,
+                &prices_store,
+            )),
+            _ => {}
+        }
+    }
+
+    Ok(EntityChanges { entity_changes })
+}
+
+fn get_event_id(event: &Event) -> String {
+    [event.hash.clone(), event.log_index.to_string()].join("-")
+}
+
+fn calculate_event_amount_usd(
+    ordinal: u64,
+    input_tokens: Vec<Erc20Token>,
+    amounts: Vec<String>,
+    store: &StoreGetBigDecimal,
+) -> BigDecimal {
+    let mut amount_usd = BigDecimal::zero();
+
+    for (idx, token) in input_tokens.iter().enumerate() {
+        let token_amount = BigInt::try_from(amounts[idx].clone()).unwrap();
+        let token_price = utils::get_token_price(ordinal, store, &token.address);
+
+        amount_usd = amount_usd + token_price.mul(token_amount.to_decimal(token.decimals))
+    }
+
+    amount_usd
+}
+
+fn create_deposit_transaction(
+    ordinal: u64,
+    pool: &Pool,
+    event: &Event,
+    deposit: &DepositEvent,
+    prices_store: &StoreGetBigDecimal,
+) -> EntityChange {
+    let id = get_event_id(event);
+
+    let mut deposit_entity_change =
+        EntityChange::new("Deposit", id.as_str(), ordinal, Operation::Create);
+
+    let input_tokens = pool.input_tokens.clone().unwrap().items;
+    let input_token_amounts = deposit.input_token_amounts.clone();
+    let output_token_amount =
+        BigInt::try_from(deposit.output_token_amount.clone().unwrap()).unwrap();
+
+    let amount_usd = calculate_event_amount_usd(
+        ordinal,
+        input_tokens,
+        input_token_amounts.clone(),
+        prices_store,
+    );
+
+    deposit_entity_change
+        .change("id", id)
+        .change("hash", event.clone().hash)
+        .change("logIndex", event.log_index as i32)
+        .change("protocol", constants::UNISWAP_V2_FACTORY.to_string())
+        .change("to", event.clone().to)
+        .change("from", event.clone().from)
+        .change("blockNumber", BigInt::from(event.block_number))
+        .change("timestamp", BigInt::from(event.timestamp))
+        .change("inputTokens", pool.input_tokens())
+        .change("outputToken", pool.output_token_address())
+        .change("inputTokenAmounts", input_token_amounts)
+        .change("outputTokenAmount", output_token_amount)
+        .change("amountUSD", amount_usd)
+        .change("pool", pool.address.clone());
+
+    deposit_entity_change
+}
+
+fn create_withdraw_transaction(
+    ordinal: u64,
+    pool: &Pool,
+    event: &Event,
+    withdraw: &WithdrawEvent,
+    prices_store: &StoreGetBigDecimal,
+) -> EntityChange {
+    let id = get_event_id(event);
+
+    let mut withdraw_entity_change: EntityChange =
+        EntityChange::new("Withdraw", id.as_str(), ordinal, Operation::Create);
+
+    let input_tokens = pool.input_tokens.clone().unwrap().items;
+    let input_token_amounts = withdraw.input_token_amounts.clone();
+    let output_token_amount =
+        BigInt::try_from(withdraw.output_token_amount.clone().unwrap()).unwrap();
```

Comment:
`withdraw.output_token_amount.as_ref().unwrap().clone()` and maybe also a from here too?